src/main/java/com/conveyal/gtfs/graphql/fetchers/PolylineFetcher.java

```diff
@@ -36,7 +45,7 @@ public Object get(DataFetchingEnvironment environment) {
             connection = GTFSGraphQL.getConnection();
             Statement statement = connection.createStatement();
             String sql = String.format(
-                "select shape_id, shape_pt_lon, shape_pt_lat from %s.shapes order by shape_id",
+                "select shape_id, shape_pt_lon, shape_pt_lat from %s.shapes order by shape_id, shape_pt_sequence",
```

Comment:
If 4 seconds is tolerable for now, then this is fine to keep, but this single query across all records could be some good low hanging fruit to refactor since it likely performs an excessive amount of sorting. 

Let's say there are 100 shapes. Let's say each shape has 500 points. 

This current implementation would sort all 100 shapes and the 500 shape points within each, so 50,000 items to sort once. Sorting takes n log<sub>2</sub>(n) time, so 1 * 50000 * log<sub>2</sub>(50000) = 780k operations. 

If there were 100 separate queries, but each only sorted 500 items at a time, this is substantially less calculation. 100 * 500 * log<sub>2</sub>(500) = 448k operations.

Reply:
There are actually 719k lines in the TriMet shapes.txt file with 1,909 unique shape IDs (so about 377 points per shape on average). With that, the numbers become:

- 1 * 719k * log<sub>2</sub>(719k) = 13.9m operations
- 1909 * 377 * log<sub>2</sub>(377) = 6.1m operations

Reply:
I just tried splitting up the queries into a `select distinct shape_id` and then iterating over the ids with a `select shape_id, shape_pt_lon, shape_pt_lat from %s.shapes where shape_id = ? order by shape_pt_sequence` and it appears to take longer for the NL feed (between 14-17 seconds). This is kind of what I was trying to get at previously: this is such a common SQL operation (a query that orders by two indexed fields) that it's probably already sufficiently optimized internally better than any kind of fanciness we could put on top of it.